Modules/cmathmodule.c

```diff
@@ -888,8 +897,21 @@ cmath_log_impl(PyObject *module, Py_complex x, PyObject *y_obj)
         y = c_log(y);
         x = _Py_c_quot(x, y);
     }
-    if (errno != 0)
-        return math_error();
+    if (errno) {
+        PyObject *ret = math_error();
+
+        if (errno == EDOM) {
```

Comment:
`errno` can be changed in `math_error()`. Either save its value in a local variable, or check `errno == EDOM` before calling `math_error()`.